packages/x-data-grid-headless/src/columnDef/columnDef.ts

```diff
@@ -0,0 +1,11 @@
+export interface ColumnDef<TData> {
+  id: string;
+  field?: keyof TData;
+  header?: string;
+  sortable?: boolean;
+  filterable?: boolean;
+  width?: number;
+  minWidth?: number;
+  maxWidth?: number;
+  flex?: number;
```

Comment:
Why are these needed? Can't they be part of a metadata? They are purely UI related, have nothing to do with the table data state & management, no?